File: Web/src/api/userCTF.js
import {apiClient, handleResponse, handleError} from "./api.js"

function GetUserCTFList(data) {
  return apiClient.post("/userCTF/GetUserCTFList", data).then(handleResponse).catch(handleError)
}

function GetVisibleUserCTFList(data) {
  return apiClient.post("/userCTF/GetVisibleUserCTFList", data).then(handleResponse).catch(handleError)
}

function AddUserCTF(data) {
  return apiClient.post("/userCTF/AddUserCTF", data).then(handleResponse).catch(handleError)
}

function RemoveUserCTF(data) {
  return apiClient.post("/userCTF/RemoveUserCTF", data).then(handleResponse).catch(handleError)
}

const userCTFApi = {
  GetUserCTFList,
  GetVisibleUserCTFList,
  AddUserCTF,
  RemoveUserCTF
}

export default userCTFApi

File: Server/route/userCTF.go
package route

import (
	"github.com/CoolCoolTomato/MatrilxArena/Server/api"
	"github.com/CoolCoolTomato/MatrilxArena/Server/middleware"
	"github.com/gin-gonic/gin"
)

func SetUserCTFRoute(r *gin.Engine) {
	userCTF := r.Group("/userCTF")
	userCTF.Use(middleware.JWTAuthMiddleware())
	{
		userCTF.POST("/GetUserCTFList", api.GetUserCTFList)
		userCTF.POST("/GetVisibleUserCTFList", api.GetVisibleUserCTFList)
		userCTF.POST("/AddUserCTF", api.AddUserCTF)
		userCTF.POST("/RemoveUserCTF", api.RemoveUserCTF)
	}
}
